go-controller/cmd/ovnkube-trace/ovnkube-trace.go

```diff
@@ -773,8 +802,10 @@ func runOvnTraceToPod(coreclient *corev1client.CoreV1Client, restconfig *rest.Co
 		} else {
 			successString = fmt.Sprintf(`output to "%s_%s"`, srcPodInfo.NodeExternalBridgeName, srcPodInfo.NodeName)
 		}
-	} else {
+	} else if !srcPodInfo.IsInterConnect || srcPodInfo.NodeName == dstPodInfo.NodeName {
```

Comment:
oh ok so here and above, what you are doing is checking if its either non-IC or if the srcNode==dstNode which means we just need to grab the local outport on the node's switch
versus
if its multizone, then we grab the transit switch patch port as the nexthop/outport... nice

what if we have more than one node in same zone and we need to go through the cluster-router into the second node's switch in same zone? and not via transit switch?

Reply:
yes @tssurya , I didn't know about subset of nodes belong to same zone. Thanks for bringing that up. Now updated the code to handle that case with commit [922c44e](https://github.com/ovn-org/ovn-kubernetes/commit/922c44eb2cff9fbeddddec2662228f1413e8b30d), will work with @dceara for ofproto/trace and ovn-trace on remote nodes.

somehow github doesn't have latest commit [922c44e](https://github.com/ovn-org/ovn-kubernetes/commit/922c44eb2cff9fbeddddec2662228f1413e8b30d) updated, hope it would eventually happen, otherwise will do force push again.